mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -12,50 +12,21 @@
 
 #define SPEEDOMETER_FILENAME "cfg/speedometer.vdf"
 #define SPEEDOMETER_DEFAULT_FILENAME "cfg/speedometer_default.vdf"
+#define SPEEDOMETER_KEY_VISIBLE "visible"
+#define SPEEDOMETER_KEY_COLORIZE "colorize"
+#define SPEEDOMETER_KEY_UNITS "units"
 
-CON_COMMAND_F(mom_hud_speedometer_savecfg,
-              "Writes the current speedometer setup for the current gamemode to file.\n"
-              "Optionally takes in the gamemode to write to.\n",
+CON_COMMAND_F(mom_hud_speedometer_loadcfg, "Loads the speedometer setup for the current gamemode from file.\n",
               FLAG_HUD_CVAR | FCVAR_CLIENTCMD_CAN_EXECUTE)
 {
-    if (args.ArgC() > 1)
-    {
-        int iGamemodeIndex = Q_atoi(args[1]);
-        if (iGamemodeIndex >= 0 && iGamemodeIndex < GAMEMODE_COUNT)
-        {
-            g_pSpeedometerData->SaveGamemodeData(GameMode_t(iGamemodeIndex));
-        }
-    }
-    else
-    {
-        g_pSpeedometerData->SaveGamemodeData(g_pGameModeSystem->GetGameMode()->GetType());
-    }
-}
-
-CON_COMMAND_F(mom_hud_speedometer_loadcfg,
-              "Loads the speedometer setup for the current gamemode from file.\n"
-              "Optionally takes in the gamemode to load from.\n",
-              FLAG_HUD_CVAR | FCVAR_CLIENTCMD_CAN_EXECUTE)
-{
-    if (args.ArgC() > 1)
-    {
-        int iGamemodeIndex = Q_atoi(args[1]);
-        if (iGamemodeIndex >= 0 && iGamemodeIndex < GAMEMODE_COUNT)
-        {
-            g_pSpeedometerData->LoadGamemodeData(GameMode_t(iGamemodeIndex));
-        }
-    }
-    else
-    {
-        g_pSpeedometerData->LoadGamemodeData(g_pGameModeSystem->GetGameMode()->GetType());
-    }
+    g_pSpeedometerData->Load(true);
 }
 
 SpeedometerData::SpeedometerData()
-    : m_pGamemodeSetupData(nullptr), m_CurrentlyLoadedGamemodeSettings(GAMEMODE_UNKNOWN)
+    : m_pSpeedoData(nullptr)
 {
     m_pDefaultSpeedoData = new KeyValues("DefaultData");
-    m_pGamemodeSetupData = new KeyValues("LoadedData");
+    m_pSpeedoData = new KeyValues("LoadedData");
 }
```

Comment:
You initialize the `m_pSpeedoData` to both nullptr and an object here, is the nullptr one no longer needed?

Reply:
oops!